compiler/rustc_ast/src/attr/mod.rs

```diff
@@ -1,10 +1,9 @@
 //! Functions dealing with attributes and meta items.
 
-use crate::ast;
-use crate::ast::{AttrId, AttrItem, AttrKind, AttrStyle, Attribute};
-use crate::ast::{Lit, LitKind};
-use crate::ast::{MacArgs, MacArgsEq, MacDelimiter, MetaItem, MetaItemKind, NestedMetaItem};
-use crate::ast::{Path, PathSegment};
+use crate::ast::{
+    self, AttrArgs, AttrArgsEq, AttrId, AttrItem, AttrKind, AttrStyle, Attribute, DelimArgs, Lit,
+    LitKind, MacDelimiter, MetaItem, MetaItemKind, NestedMetaItem, Path, PathSegment,
+};
```

Comment:
Could you keep the single-line style for imports, here and in other files?

Reply:
The great thing about rustfmt is that it mostly avoids the need to make arbitrary formatting decisions. But here:
```
use crate::ast::{AttrId, AttrItem, AttrKind, AttrStyle, Attribute};
use crate::ast::{Lit, LitKind};
use crate::ast::{MacArgs, MacArgsEq, MacDelimiter, MetaItem, MetaItemKind, NestedMetaItem};
```
I have multiple choices: add to the end of the first line, add to the start of the second line, or even combine the first and second line. And this is a relatively clean example where everything is in alphabetical order. We have lots of examples like this one:
```
use rustc_ast::{attr, BindingAnnotation, ByRef, Term};
use rustc_ast::{GenericArg, MacArgs, MacArgsEq};
use rustc_ast::{GenericBound, SelfKind, TraitBoundModifier};
```
where alphabetical order is not maintained across lines. Where should I insert an identifier like `DelimArgs` here? It could be on any of the lines. That's exactly the kind of arbitrary choice I don't like having to make. The multi-line style avoids this, because rustfmt does it all for you. I think that's why it gets used for cases like [this](https://github.com/rust-lang/rust/blob/911cbf8e46f0f3b044c87fc5cbca922878db757b/compiler/rustc_parse/src/parser/expr.rs#L8-L26).

Having said that, I'll revert if it's necessary to get r+.

Reply:
>Where should I insert an identifier like `DelimArgs` here? 

Same as with multi-line imports, into its alphabetic order position.

>I think that's why it gets used for cases like [this](https://github.com/rust-lang/rust/blob/911cbf8e46f0f3b044c87fc5cbca922878db757b/compiler/rustc_parse/src/parser/expr.rs#L8-L26).

For cases like that I'd personally use a glob import.

Reply:
Single-line imports are easily greppable, and that's something I do quite regularly, that's why I keep this style in crates that I review and that already use this style.